packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,180 @@
+
+### ARIA Treegrid guidance
+
+Some noteworthy aspects of the treegrid pattern are:
+
+1. All cells are focusable, even those that have no interactive elements. The left and right arrow keys, while in navigation modes, move focus from cell to cell. In the case where a cell has a single interactive element, the element can be given the focus instead of the containing cell.
+
+> ARIA: "One reason it is important for all cells to be able to receive or contain keyboard focus is that screen readers will typically be in their application reading mode, rather than their document reading mode, when users are interacting with the grid. While in application mode, a screen reader user hears only focusable elements and content that labels focusable elements. So, screen reader users may unknowingly overlook elements contained in a treegrid that are either not focusable or not used to label a column or row."
+
+2. The `Tab` key moves focus between the focusable elements in a row (not the cells). When focus has reached the end (or beginning of the row), a continued `Tab` (or `Shift-Tab`) will move focus outside of the table.
+3. When a cell has multiple interactive elements, or editable content, the user needs a way to leave navigation mode to interact with the cell (`Enter` and `F2` are suggested), and a way to return to navigation mode (`Esc`, also `Enter` for editable content).  
+   (Note: The `treegrid` and `grid` guidance are identical here.)
+
+If the above guidelines are adhered to, technologies like screen readers should be able to handle navigating through the table in an expected fashion.
+
+There's also guidance about whether cells themselves, or the elements inside the cells, should be focused:
+
+> ARIA: "For assistive technology users, the quality of experience when navigating a grid heavily depends on both what a cell contains and on where keyboard focus is set. For example, if a cell contains a button and a grid navigation key places focus on the cell instead of the button, screen readers announce the button label but do not tell users a button is present.  
+> There are two optimal cell design and focus behavior combinations:
+>
+> -   A cell contains one widget whose operation does not require arrow keys and grid navigation keys set focus on that widget. Examples of such widgets include link, button, menubutton, toggle button, radio button (not radio group), switch, and checkbox.
+> -   A cell contains text or a single graphic and grid navigation keys set focus on the cell."
+
+### Specifics for nimble-table
+
+#### Focusing the table, `Tab`/`Shift-Tab` behavior
+
+`Tab` or `Shift-Tab`, from neighboring elements on the page (according to tab order), will focus the table:
+
+-   If the table header contains any interactive/focusable elements, the 1st will be focused. Examples: 'Select all rows' checkbox, 'Collapse all groups' button
```

Comment:
Will a column header itself be focusable? For example, in order to toggle the sort order of the column.

Reply:
This probably needs some discussion (I can bring it up at the team meeting tomorrow).

I wasn't originally thinking of making the column headers focusable, but that would mean that there wouldn't be a keyboard-accessible way of sorting columns, since we deprioritized implementing the column header menus. So maybe we should.

Reply:
As discussed at team meeting, added a section for column sorting.

> #### Sorting
> If at least one column can be sorted, all of the column headers will be keyboard focusable (the [ARIA grid examples](https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/#examples) illustrate this idea), to allow for column sorting via the keyboard.
> In this scenario:
> -   `ArrowLeft` and `ArrowRight` will navigate between the selection checkbox and expand/collapse button, and each of the column headers, following similar semantics as the cell navigation (above).
> -   `Enter` will toggle column sorting (between ascending/descending/none, in the same way a mouse click does)
> -   `Shift-Enter` will add/remove the given column from sorting, while maintaining any other sorted columns (e.g. multi-sort), in the same way a Shift-mouse-click does